server/plugins/builder/pages/index.js

```diff
@@ -25,33 +25,39 @@ class Page extends EngineBasePage {
         onPreHandler: {
           method: async (request, h) => {
             let files = fileStreamsFromPayload(request.payload)
-            if (!files.length) {
-              return h.continue
-            }
-            let file = files[0]
-            let key = file[0]
-            let previousUpload = (request.yar.get('originalFilenames') || {})[key]
-            if (previousUpload && file[1].hapi.filename === '') {
-              h.request.payload[key] = previousUpload.location
+            files.forEach(file => {
+              if (file[1]._data.length < 1) {
+                delete request.payload[file[0]]
```

Comment:
Do we neede to delete the payload value? Won't it just be ignored, or is it validated later?

Reply:
it's validated later `onPostHandler` from engine's page.js. Since the buffer is still coming through even though its .length === 0 validation doesn't pick up that it's empty so manually deleting here and doing h.continue